src/platform/mt8195/lib/dma.c

```diff
@@ -0,0 +1,60 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2021 Mediatek
+//
+// Author: YC Hung <yc.hung@mediatek.com>
+
+#include <sof/common.h>
+#include <sof/drivers/interrupt.h>
+#include <sof/lib/dma.h>
+#include <sof/lib/memory.h>
+#include <sof/sof.h>
+#include <sof/spinlock.h>
+
+#include <mt8195-afe-regs.h>
+#include <mt8195-afe-common.h>
+
+extern struct dma_ops dummy_dma_ops;
+extern struct dma_ops memif_ops;
+
+SHARED_DATA struct dma dma[PLATFORM_NUM_DMACS] = {
+{
+	.plat_data = {
+		.id		= DMA_ID_HOST,
+		.dir		= DMA_DIR_HMEM_TO_LMEM | DMA_DIR_LMEM_TO_HMEM,
+		.devs		= DMA_DEV_HOST,
+		.channels	= 16,
+	},
+	.ops	= &dummy_dma_ops,
+},
```

Comment:
Is you physical DRAM accessible both from Application Processor (Linux kernel) and also from DSP (SOF firmware)? We have introduced the Dummy DMA just because of this. 

We wanted to fit into SOF flow, but somehow this copying is not really needed if the DRAM is shared. We need to optimize  that in the future.